### ISSUE TITLE
Search in permissions is broken

### ISSUE THREAD

**Issue by ansiwen** [RocketChat/Rocket.Chat#28227]
### Description:

Under `admin/permissions` the search for  "Permissions" and "Settings" is very unreliable. For example words like "screen", "two", "private", ... all result into "no data found" 

### Steps to reproduce:

1. Go to `.../admin/permissions`
2. Enter "screen" in the search field

### Expected behavior:

See all items that include the word "screen"

### Actual behavior:

"no data found" is displayed

### Server Setup Information:

- Version of Rocket.Chat Server: 5.4.3
- Operating System: Ubuntu
- Deployment Method: snap
- Number of Running Instances: 1
- DB Replicaset Oplog: rs0
- NodeJS Version: v14.19.3
- MongoDB Version: 4.4.15 / wiredTiger (oplog Enabled)

### Client Setup Information

- Desktop App or Browser Version: Google Chrome 110.0.5481.177
- Operating System: MacOS 12.6.1

**Comment by dudanogueira** [comment]
I was able to reproduce this on latest 6.0.0.rc-3

**Comment by manulpatel** [comment]
Hello @ansiwen  @dudanogueira. I am new to the commnity. Could you please guide me on the next steps to work in this?

**Comment by MohorBabai** [comment]
@dudanogueira Yes how to start contributing as a backend developer?

**Comment by destrex271** [comment]
Yup was able to reproduce this issue, would be looking into a fix for the same.

**Comment by destrex271** [comment]
The problem is occuring because the parameter in the search textfield is querying the id of the permission item and for permissions with "screen" in them the id does not contain the keyword as seen in the screenshot below:

![Screenshot from 2023-03-02 22-44-32](https://user-images.githubusercontent.com/59926121/222520295-f8b5c602-e9f4-4e26-b602-240b80280ee4.png)

Therefore we either need to change the query parameter or just fix the two entries by adding the screen keyword in their id. I am preparing a pr for the same but am facing some issues while pushing to my fork : https://pastebin.com/cHh5q5J2
If anyone has any pointers on how to fix this please guide me.

**Comment by krsuumar** [comment]
Hi,

As per the query, searching is based on the _id key. Attached below screenshot.
<img width="378" alt="image" src="https://user-images.githubusercontent.com/25521962/222946771-bf6cc6b1-462e-4723-8b9c-1e89ce24c622.png">

And then _id is translated using en.json file which we are seeing on the "Permissions" page.
**For example** "access-mailer" is stored as _id value in database table and then it is translated into "Access Mailer Screen" using en.json file.
Attached below screenshot.

1. _id: "access-mailer" stored in database
    
<img width="583" alt="image" src="https://user-images.githubusercontent.com/25521962/222947485-c6e25fca-a99b-4b12-a558-7fe233ed2e23.png">

2. "access-mailer" translated to "Access Mailer Screen" using en.json file.
 
<img width="583" alt="image" src="https://user-images.githubusercontent.com/25521962/222947626-28427c01-cc0b-4108-8f95-78bd28a39ea5.png">

3. Hence  "Access Mailer Screen" is visible on page.

<img width="732" alt="image" src="https://user-images.githubusercontent.com/25521962/222947701-da54ef90-573f-41c0-9694-9ccd9f979a22.png">


**Problem** here is that "screen" word is not present in the _id column of database, so using "screen" as search keyword giving no result.

Attaching the screenshot of the query **{_id:{$regex:'screen',$options:"$i"},level:{$ne:'settings'}}**.
<img width="617" alt="image" src="https://user-images.githubusercontent.com/25521962/222947906-ee7bdf2b-dbd5-4cd9-9a5a-84eb36f25d2c.png">



As per my understanding, there is two way for solving this issue.
1. Update the database table with same value as shown in the "Permissions" Page. But not sure whether it may break the other files processing.
2. Update the en.json file with the same value as stored in the _id column of database. But also not sure whether it may break other files processing as well.

@dudanogueira please look into this and guide me.

Thanks

**Comment by Anil-Kumar-Git** [comment]
Now all data is properly filtered .

https://user-images.githubusercontent.com/109125942/225015237-263959ca-c0b1-410a-97e0-42b56e67df99.mp4

**Comment by shoebilyas123** [comment]
@XpertWebApp the issue is the data stored in the backend. Did you edit the front end only? That might break the feature if you altered the useLanguage. If you altered the mongo values then it might break the application at other places. You must make sure it doesn't.

Also @krsuumar, where is the en.json file located?

**Comment by krsuumar** [comment]
> @XpertWebApp the issue is the data stored in the backend. Did you edit the front end only? That might break the feature if you altered the useLanguage. If you altered the mongo values then it might break the application at other places. You must make sure it doesn't.
> 
> Also @krsuumar, where is the en.json file located?

 Rocket.Chat/packages/ui-contexts/src/en.json

**Comment by mko-x** [comment]
On search you need to look in the translated strings from the current language *.json so that the search works in any language and then only show the related ids.

**Comment by Aditya-eddy** [comment]
@ansiwen is this issue resolved?

**Comment by Rbcoder1** [comment]
If This Issue not solved I Am Interested to work it
can you please assign it to me

**Comment by SyedAbuBakerAli** [comment]
hi, @ansiwen, can you assign me this issue.

**Comment by Abhii-Agarwal09** [comment]
Do we have any sample test accounts? Or should we create our own account and organization and test?

**Comment by Vandit34** [comment]
👋🏻hi @kh-ac I saw your proposed solution & it looks great.
I wanted to know from you  & other contributors of this issue 

I was thinking we can implement that CTRL + F type functionality: as the user types the field name in the search box then they will get it after the search is over. Correct me if this idea is good or bad ?

(❁´◡`❁)  By the way, I am a newbie in the open-source environment because I started the journey last month

**Comment by JimmyKurui** [comment]
Can i work on this too if this issue has not yet been submitted?

**Comment by IITI-tushar** [comment]
Hi
Is this issue still active? I would like to work on this issue.

**Comment by reetp** [comment]
Referred to team for comments.

**Comment by reetp** [comment]
Understand this is already in hand and slated for release with 7.3.x

For those doing PRs please *always* ask here before starting any work. It will save time and frustration.


https://open.rocket.chat/channel/support

**Comment by casalsgh** [comment]
Internal team will be checking into this and will get back
